### ISSUE TITLE
deploy logging project: manifest not found

### ISSUE THREAD

**Issue by lorenzvth7** [openshift/origin#9745]
We are using our own docker registry which is secured with a selfsigned certificate. So if we place the cert on our openshift node we're able to pull. We pulled the openshift-origin images v1.2.0 from dockerhub and pushed it inside our docker registry. We are using the registry instead of docker.io/openshift/origin-xxx 
This works fine for our router, registry, cluster metrics project etc..

But when we are deploying the logging project: https://docs.openshift.org/latest/install_config/aggregate_logging.html it doesn't work.
the pull of the `registry.com/asco/origin-logging-deployment:v1.2.0` is fine and it deploys.
But the problem raises later. The fluentd image is also pulled fine (from our registry).
But the images of the rest aren't pulled in the right way, 

```

logging-auth-proxy  172.30.xx:5000/logging/logging-auth-proxy   latest, v1.2.0  29 minutes ago
logging-curator     172.30.6xx:5000/logging/logging-curator     latest, v1.2.0  29 minutes ago
logging-elasticsearch   172.30.xx:5000/logging/logging-elasticsearch    latest, v1.2.0  29 minutes ago
logging-fluentd     172.30.xx/logging/logging-fluentd   latest, v1.2.0  29 minutes ago
logging-kibana  172.30.xx:5000/logging/logging-kibana   latest, v1.2.0  29 minutes ago
```

The other images are describing a`@sha256:xxx` manifest after the image name. The images in our registry have an _manifest folder which contains other sha(s).
##### Version

```
oc v1.2.0
kubernetes v1.2.0-36-g4a3f9c5
```
##### Steps To Reproduce

Following official documenation + replacing registry to our registry
##### Current Result

`pulling image "registry.com/asco/origin-logging-auth-proxy@sha256:179b84eb803fac116f913182c2fb64a2e7adf01dd04fc58e1336d96ce0ce3d65"`

`Failed to pull image "registry.com/asco/origin-logging-auth-proxy@sha256:179b84xb803fac116f913182c2fb64a2e7adf01dd04fc58e1336d96ce0ce3d65": image pull failed for registry.com/asco/origin-logging-auth-proxy@sha256:179b84ex803fac116f913182c2fb64a2e7adf01dd04fc58e1336d96ce0ce3d65, this may be because there are no credentials on this request. details: (manifest unknown: manifest unknown)`

example of a sha which isn't in our docker registry:
`sha256:179b84eb803fac116f913182c2fb64a2e7adf01dd04fc58e1336d96ce0ce3d6`

We've pulled the images on a machine with docker verison 1.10.3 (yum install docker). Just like on our cluster.

**Comment by sosiouxme** [comment]
Reference conversation on dev list: https://lists.openshift.redhat.com/openshift-archives/users/2016-July/msg00029.html

It's not clear to me what's going on here but it's something docker-related, not specific to the logging images. I would be surprised if you did not have the same problem deploying from other images in the registry.

When we've seen this problem in the past it had to do with docker 1.10 pushing an image that docker 1.9 could not read, so double-check your docker versions. But this could be a different problem with 1.10. I would advise continuing the discussion on dev-list as it's the most visible.

**Comment by lorenzvth7** [comment]
@sosiouxme I don't have the same issue for other images. I'm working on v1.2.0 but only recently. This cluster has docker 1.10.3 To be sure I even pulled the images (of origin-kibana etc) on my master of that cluster (with also docker 1.10.3). Tagged them and pushed the images to my own Docker Registry (not the openshift registry). And than started the deploy with images from that registry.

All the other images like the metrics project, router, registry, ... are pulled in a good way and are running fine.

**Comment by lorenzvth7** [comment]
Images are really the same but the `sha` manifest is different 
```
registry.xxx.com/proj/origin-logging-curator            v1.2.0              3bdf7e898916        11 weeks ago        225.9 MB
docker.io/openshift/origin-logging-curator               <none>              3bdf7e898916        11 weeks ago        225.9 MB
```

**Comment by lorenzvth7** [comment]
pull original image
```
$ docker pull docker.io/openshift/origin-logging-curator:v1.2.0
...
Digest: sha256:bb8e1038ec218fa32f332a5ce4cbc1943a2698e529d43e582979328e9ea0625a
```
tag image on our registry
`$ docker tag docker.io/openshift/origin-logging-curator@sha256:bb8e1038ec218fa32f332a5ce4cbc1943a2698e529d43e582979328e9ea0625a registry.xxx.com/proj/origin-logging-curator:v1.2.0`

push + digest is different
```
$ docker push registry.evolane-ops.be/evolane/origin-logging-curator:v1.2.0
The push refers to a repository [registry.evolane-ops.be/evolane/origin-logging-curator]
5f70bf18a086: Mounted from evolane/origin-pod
49fa9df8b12a: Pushed
58a3067284b2: Pushed
6a6c96337be1: Mounted from evolane/origin-metrics-deployer
v1.2.0: digest: sha256:f5cc3af920595938eb1e81f8aea08db4230d37d529a4748620846a240364651b size: 2780
```

**Comment by lorenzvth7** [comment]
Still have this issue.
E.g. [here](https://hub.docker.com/r/openshift/origin-logging-fluentd/tags/) are tags of the origin images of the EKF stack. So when origin 1.3 will be released I'm afraid we will lose the 1.2 tag in docker hub. So when the image isn't locally on our server we can't pull it anymore and we need to use 1.3 on an 1.2 environment. That's the reason we try to 'fork' the images. Workes for metrics, deployers etc but nog for the images of the logging EFK project

**Comment by jcantrill** [comment]
@lorenzvth7 The 1.3 images were released.  Can you confirm if you are still experiencing this issue?
